pysb/core.py

```diff
@@ -1671,6 +1689,32 @@ def rename(self, c, new_name):
             del m[c.name]
 
 
+class OdeView(collections.Sequence):
+    """Compatibility shim for the Model.odes property."""
+
+    # This is necessarily coupled pretty tightly with Model. Note that we
+    # faithfully emulate the detail of the original implementation in which odes
+    # is an empty list before the equation generation process is run (and after
+    # reset_equations is called). Now the "empty" condition is when species is
+    # empty.
+
+    def __init__(self, model):
+        self.model = model
+
+    def __getitem__(self, key):
+        if not self.model.species:
+            raise IndexError('list index out of range')
+        if isinstance(key, slice):
+            return [self[k] for k in range(*key.indices(len(self)))]
+        sr = self.model.stoichiometry_matrix[key]
+        terms = [self.model.reactions[i]['rate'] * v
+                 for i, v in zip(sr.indices, sr.data)]
```

Comment:
I've just submitted PR #294, which replaces Python RHS `eval` with `lambdify` and has experimental (optional, off by default) theano support in ScipyOdeSimulator. It's 30% faster on the time_scipy_lsoda benchmark for me (compared with current master 553eadc) and the above change seems to make it slightly faster, not slower, if that helps.